crates/codegen/runtime/cargo/wasm/src/runtime/interface/cst.wit.jinja2

```diff
@@ -51,108 +68,184 @@ interface cst {
+        /// A terminal node that represents a recognized part of the source string.
         terminal(terminal-node)
     }
 
+    /// Represents a non-terminal node in the syntax tree.
+    /// These nodes can have child nodes and represent language constructs.
     resource nonterminal-node {
+        /// Returns a unique numerical identifier of the node.
+        /// It is only valid for the lifetime of the enclosing tree.
+        /// It can change between multiple parses, even for the same source code input.
         id: func() -> u32;
+        /// Returns the kind enum of this nonterminal node.
         kind: func() -> nonterminal-kind;
+        /// Returns the length of the text span this node covers.
         text-length: func() -> text-index;
 
+        /// Returns the list of child edges connected to this node.
         children: func() -> list<edge>;
+        /// Converts the node and its children back to source code text.
         unparse: func() -> string;
+        /// Converts the node to a JSON representation for debugging.
         to-json: func() -> string;
 
+        /// Creates a cursor positioned at the given text offset within this node.
         create-cursor: func(text-offset: text-index) -> cursor;
     }
 
+    /// Represents a terminal node in the syntax tree.
+    /// These are leaf nodes that represent actual tokens from the source code.
     resource terminal-node {
+        /// Returns a unique numerical identifier of the node.
+        /// It is only valid for the lifetime of the enclosing tree.
+        /// It can change between multiple parses, even for the same source code input.
         id: func() -> u32;
+        /// Returns the kind enum of this terminal node.
         kind: func() -> terminal-kind;
+        /// Returns the length of the text span this node covers.
         text-length: func() -> text-index;
 
+        /// Returns the list of child edges connected to this node.
         children: func() -> list<edge>;
+        /// Converts the node back to source code text.
         unparse: func() -> string;
+        /// Converts the node to a JSON representation for debugging.
         to-json: func() -> string;
     }
 
+    /// Represents a connection between nodes in the syntax tree.
     record edge {
+        /// Optional label describing the relationship between nodes.
         label: option<edge-label>,
+        /// The target node of this edge.
         node: node,
     }
 
+    /// Provides navigation and traversal capabilities over the syntax tree.
     resource cursor {
+        /// Resets the cursor to its initial position.
         reset: func();
+        /// Marks the cursor as completed.
         complete: func();
+        /// Returns whether the cursor has completed its traversal.
         is-completed: func() -> bool;
 
+        /// Creates a copy of this cursor at its current position.
         clone: func() -> cursor;
+        /// Creates a new cursor at the same position as this one.
```

Comment:
, but without the connection to it's ancestors, so iteration through it will only cover this subtree.